phoenix/app/apiProducts/lib/apiProducts_web/plugs/plug_id.ex

```diff
@@ -0,0 +1,23 @@
+defmodule ApiProductsWeb.Plugs.PlugId do
+  import Plug.Conn
+
+  alias ApiProducts.Catalog
+  
+  # plug for find product by id
+  defp find_product(conn, id) do
+    id = conn.params["id"]
+    product = Catalog.get_product(id)
+    if product do 
+      assign(conn, :product, product)
+    else 
+      send_resp(conn, 404, "Product not found")
```

Comment:
Aqui faltou dar halt na conn, para evitar de enviar duas respostas para o mesmo request que causa o erro:
(Plug.Conn.AlreadySentError) the response was already sent.